### PR TITLE
Story/io 233/summary table monthly view

### PR DESCRIPTION
**Pull request by EngbergJonas**
- Added monthly view to each year column in the PlanningViewSummary table
  - Added graph for each year in the monthly view that shows how much of each month is planning and construction
  - Added a line that shows the current date when the monthly view is expanded
- Changed ProjectCells to take planningStartYear and constructionEndYear into account when creating the timeline
- Changed ProjectBasicsForm to also patch the planningStartYear and constructionEndYear when estPlanningStart or estConstructionEnd is given or patched

### CODE REVIEW

**Review by kgunga — CHANGES_REQUESTED** [review]

File: src/components/Planning/PlanningTable/PlanningRow/ProjectRow/ProjectYearSummary/MonthlyGraphCell.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  return (
+    <td
+      className={`monthly-cell project ${cellType} ${cellColor}`}
+      data-testid={`project-monthly-graph-cell-${id}-${month}`}
+    >
+      <div className={`monthly-graph-cell-container ${cellType}`}>
+        <>
+          {/* Planning bar */}
+          <div
+            className={`monthly-planning-bar-container ${planning.isStart ? 'justify-end' : ''}`}
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Planning/PlanningTable/PlanningRow/ProjectRow/ProjectYearSummary/MonthlyGraphCell.tsx]: Couldn't be done with the current element structure, added a comment...

File: src/components/Planning/PlanningTable/PlanningRow/ProjectRow/ProjectYearSummary/MonthlyGraphCell.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+const MonthlyGraphCell: FC<IMonthlyGraphCellProps> = ({
+  index,
+  planning,
+  construction,
+  cellType,
+  month,
+  id,
+}) => {
+  const cellColor = useMemo(() => (index % 2 == 0 ? 'bg-bus-l' : 'bg-white'), [index]);
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Planning/PlanningTable/PlanningRow/ProjectRow/ProjectYearSummary/MonthlyGraphCell.tsx]: fixed

File: src/components/Planning/PlanningTable/PlanningRow/PlanningCell.tsx [hunk 3]
```diff
@@ -1,31 +1,46 @@
...
+      {year === selectedYear && (
+        <>
+          {isCurrentYear && (
+            <td key={`${year}-monthly-cell`} className={`monthly-summary-cell ${type}`}>
+              {/* TODO: some stuff here */}
+            </td>
+          )}
+          {moment.monthsShort().map((m) => (
+            <td key={m} className={`monthly-cell ${type}`}>
+              {/* TODO: some stuff here */}
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Planning/PlanningTable/PlanningRow/PlanningCell.tsx]: fixed

File: src/components/Planning/PlanningTable/PlanningRow/PlanningCell.tsx [hunk 4]
```diff
@@ -1,31 +1,46 @@
...
+          <span data-testid={`deviation-${id}-${year}`} className="planning-cell-deviation">
+            {deviation}
+          </span>
+        </div>
+      </td>
+      {year === selectedYear && (
+        <>
+          {isCurrentYear && (
+            <td key={`${year}-monthly-cell`} className={`monthly-summary-cell ${type}`}>
+              {/* TODO: some stuff here */}
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Planning/PlanningTable/PlanningRow/PlanningCell.tsx]: fixed

File: src/components/Planning/PlanningSummaryTable/PlanningSummaryTableHeadCell.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      const daysInMonth = getDaysInMonthForYear(year, month);
+      const dayToday = parseInt(moment().format('D'));
+      const percentOfMonthThatHasPast = calcPercentage(dayToday, daysInMonth);
+
+      return `${Math.floor(percentOfMonthThatHasPast / 2.55)}px`;
+    },
+    [year],
+  );
+
+  const leftArrow = useMemo(
```
[the remark this answers is not in the document]
Comment (EngbergJonas) [reply src/components/Planning/PlanningSummaryTable/PlanningSummaryTableHeadCell.tsx]: fixed

**Review by kgunga — APPROVED** [review]
👍

**Pull request merged**